Run UR Simulator from users home directory.
Index: ursim/start-ursim.sh
===================================================================
--- ursim.orig/start-ursim.sh
+++ ursim/start-ursim.sh
@@ -1,6 +1,7 @@
 #!/bin/bash
 
 export LD_LIBRARY_PATH=/opt/urtool-3.0/lib
+export PATH=/opt/ursim/5.11.11/bin/:$PATH
 
 URSIM_ROOT=$(dirname $(readlink -f $0))
 
@@ -9,7 +10,6 @@ pushd $URSIM_ROOT &>/dev/null
 CLASSPATH=$(echo $URSIM_ROOT/lib/*.jar | tr ' ' ':')
 
 ./stopurcontrol.sh
-./starturcontrol.sh
 
 ROBOT_TYPE="UR5"
 if [ "$1" == "UR10" ] ; then
@@ -22,35 +22,90 @@ fi
 
 #Setting up the configuration files according to the robot type
 #urcontrol.conf
-rm -f $URSIM_ROOT/.urcontrol/urcontrol.conf
-ln -s $URSIM_ROOT/.urcontrol/urcontrol.conf.$ROBOT_TYPE $URSIM_ROOT/.urcontrol/urcontrol.conf
+CONFIG_ROOT=$HOME/.ur
+
+mkdir -p $CONFIG_ROOT
+mkdir -p $CONFIG_ROOT/.urcontrol
+
+if [ ! -f "$CONFIG_ROOT/.urcontrol/urcontrol.conf.$ROBOT_TYPE" ]; then
+    ln -s $URSIM_ROOT/.urcontrol/urcontrol.conf.$ROBOT_TYPE $CONFIG_ROOT/.urcontrol/urcontrol.conf.$ROBOT_TYPE
+fi
+rm -f $CONFIG_ROOT/.urcontrol/urcontrol.conf
+ln -s $CONFIG_ROOT/.urcontrol/urcontrol.conf.$ROBOT_TYPE $CONFIG_ROOT/.urcontrol/urcontrol.conf
+
+if [ ! -f "$CONFIG_ROOT/.urcontrol/joint_size0.conf" ]; then
+    ln -s $URSIM_ROOT/.urcontrol/joint_size0.conf $CONFIG_ROOT/.urcontrol/joint_size0.conf
+fi
+if [ ! -f "$CONFIG_ROOT/.urcontrol/joint_size1.conf" ]; then
+    ln -s $URSIM_ROOT/.urcontrol/joint_size1.conf $CONFIG_ROOT/.urcontrol/joint_size1.conf
+fi
+if [ ! -f "$CONFIG_ROOT/.urcontrol/joint_size2.conf" ]; then
+    ln -s $URSIM_ROOT/.urcontrol/joint_size2.conf $CONFIG_ROOT/.urcontrol/joint_size2.conf
+fi
+if [ ! -f "$CONFIG_ROOT/.urcontrol/joint_size3.conf" ]; then
+    ln -s $URSIM_ROOT/.urcontrol/joint_size3.conf $CONFIG_ROOT/.urcontrol/joint_size3.conf
+fi
+if [ ! -f "$CONFIG_ROOT/.urcontrol/joint_size4.conf" ]; then
+    ln -s $URSIM_ROOT/.urcontrol/joint_size4.conf $CONFIG_ROOT/.urcontrol/joint_size4.conf
+fi
 
 #ur-serial
-rm -f $URSIM_ROOT/ur-serial
-ln -s $URSIM_ROOT/ur-serial.$ROBOT_TYPE $URSIM_ROOT/ur-serial
+if [ ! -f "$CONFIG_ROOT/ur-serial.$ROBOT_TYPE" ]; then
+    ln -s $URSIM_ROOT/ur-serial.$ROBOT_TYPE $CONFIG_ROOT/ur-serial.$ROBOT_TYPE
+fi
+rm -f $CONFIG_ROOT/ur-serial
+ln -s $CONFIG_ROOT/ur-serial.$ROBOT_TYPE $CONFIG_ROOT/ur-serial
 
 #safety.conf
-rm -f $URSIM_ROOT/.urcontrol/safety.conf
-ln -s $URSIM_ROOT/.urcontrol/safety.conf.$ROBOT_TYPE $URSIM_ROOT/.urcontrol/safety.conf
+if [ ! -f "$CONFIG_ROOT/.urcontrol/safety.conf.$ROBOT_TYPE" ]; then
+    cp $URSIM_ROOT/.urcontrol/safety.conf.$ROBOT_TYPE $CONFIG_ROOT/.urcontrol/safety.conf.$ROBOT_TYPE
+fi
+rm -f $CONFIG_ROOT/.urcontrol/safety.conf
+ln -s $CONFIG_ROOT/.urcontrol/safety.conf.$ROBOT_TYPE $CONFIG_ROOT/.urcontrol/safety.conf
 
 #program directory
-rm -f $URSIM_ROOT/programs
-ln -s $URSIM_ROOT/programs.$ROBOT_TYPE $URSIM_ROOT/programs
+mkdir -p $CONFIG_ROOT/programs.$ROBOT_TYPE
+if [ ! -f "$CONFIG_ROOT/programs.$ROBOT_TYPE/license.p7b" ]; then
+    ln -s $URSIM_ROOT/programs.$ROBOT_TYPE/license.p7b $CONFIG_ROOT/programs.$ROBOT_TYPE/license.p7b
+fi
+rm -f $CONFIG_ROOT/programs
+ln -s $CONFIG_ROOT/programs.$ROBOT_TYPE $CONFIG_ROOT/programs
+
+#GUI directory
+mkdir -p $CONFIG_ROOT/GUI
+rm -f $CONFIG_ROOT/GUI/bin
+rm -f $CONFIG_ROOT/GUI/bundle
+rm -f $CONFIG_ROOT/GUI/conf
+rm -rf $CONFIG_ROOT/GUI/felix-cache
+ln -s $URSIM_ROOT/GUI/bin $CONFIG_ROOT/GUI/bin
+ln -s $URSIM_ROOT/GUI/bundle $CONFIG_ROOT/GUI/bundle
+ln -s $URSIM_ROOT/GUI/conf $CONFIG_ROOT/GUI/conf
+
+# URControl
+if [ ! -f "$CONFIG_ROOT/stopurcontrol.sh" ]; then
+    ln -s $URSIM_ROOT/stopurcontrol.sh $CONFIG_ROOT/stopurcontrol.sh
+fi
+if [ ! -f "$CONFIG_ROOT/starturcontrol.sh" ]; then
+    ln -s $URSIM_ROOT/starturcontrol.sh $CONFIG_ROOT/starturcontrol.sh
+fi
 
 # Robot Root Certificate check
-$URSIM_ROOT/ursim-certificate-check.sh
+#$URSIM_ROOT/ursim-certificate-check.sh
+
+HOME=$CONFIG_ROOT ./starturcontrol.sh
 
 #Start the gui
-pushd GUI
-HOME=$URSIM_ROOT java -Duser.home=$URSIM_ROOT -Dconfig.path=$URSIM_ROOT/.urcontrol -Djava.library.path=/usr/lib/jni -jar bin/*.jar
+pushd $CONFIG_ROOT/GUI
+HOME=$CONFIG_ROOT /usr/lib/jvm/java-8-openjdk-amd64/bin/java -Duser.home=$CONFIG_ROOT -Dconfig.path=$CONFIG_ROOT/.urcontrol -Djava.library.path=/usr/lib/jni -jar bin/*.jar
 popd
 
-
 #clean up
-rm -f $URSIM_ROOT/.urcontrol/urcontrol.conf
-rm -f $URSIM_ROOT/ur-serial
-rm -f $URSIM_ROOT/.urcontrol/safety.conf
-rm -f $URSIM_ROOT/programs
+rm -f $CONFIG_ROOT/.urcontrol/urcontrol.conf
+rm -f $CONFIG_ROOT/ur-serial
+rm -f $CONFIG_ROOT/.urcontrol/safety.conf
+rm -f $CONFIG_ROOT/programs
+
+./stopurcontrol.sh

 popd &>/dev/null
 
Index: ursim/starturcontrol.sh
===================================================================
--- ursim.orig/starturcontrol.sh
+++ ursim/starturcontrol.sh
@@ -1,3 +1,5 @@
 #!/bin/bash
 SCRIPT_DIR=$(dirname $(readlink -f $0))
-HOME=$SCRIPT_DIR $SCRIPT_DIR/URControl -r &>$SCRIPT_DIR/URControl.log &
+CONFIG_DIR=$HOME
+mkdir -p $CONFIG_DIR/.urcontrol
+HOME=$CONFIG_DIR $SCRIPT_DIR/URControl -r &>$CONFIG_DIR/URControl.log &
Index: ursim/stopurcontrol.sh
===================================================================
--- ursim.orig/stopurcontrol.sh
+++ ursim/stopurcontrol.sh
@@ -1,2 +1,2 @@
 #!/bin/bash
-killall -9 -w URControl
+pkill -9 URControl
